### PR TITLE
fix(NODE-3150): added bsonRegExp option for v3.6

### PR DESCRIPTION
**Pull request by andymina**
## Description
- Allowed retrieving PCRE-style RegExp by adding bsonRegExp option to commands, collections, and dbs
- Added unit tests for the bsonRegExp option

### CODE REVIEW

**Review by durran — APPROVED** [review]

**Review by durran — APPROVED** [review]
Just note I did change the base branch to 3.7 from 3.6

**Review by nbbeeken — CHANGES_REQUESTED** [review]
Just leaving a note so we can remember, there's some refactoring needed for the older node versions in the test. (i.e. cant use async)

**Review by nbbeeken — APPROVED** [review]

File: test/unit/bson_regex.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+              { regex: new BSONRegExp('abc', 'imx') },
+              passOptionTo === 'operation' ? option : undefined
+            )
+          )
+          .then(res =>
+            expect(res)
+              .has.property('regex')
+              .that.is.instanceOf(BSONRegExp)
+          )
+          .then(() => client.close());
```
Comment (dariakp) [inline test/unit/bson_regex.test.js:36]: in order to make sure the test actually does close the connection even if the test fails, please include a rejection handler, too; you can pass in a done parameter in L11, so this line would become:
`.then(() => client.close(done), (err) => client.close(() => done(err))`

**Review by dariakp — CHANGES_REQUESTED** [review]

File: test/unit/bson_regex.test.js (hunk 1 above)
Comment (andymina) [reply test/unit/bson_regex.test.js]: just added!

**Review by dariakp — APPROVED** [review]

File: test/unit/bson_regex.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+describe('BSONRegExp', () => {
+  describe('bsonRegExp option', () => {
+    // define option for tests to use
+    const option = { bsonRegExp: true };
+    for (const passOptionTo of ['client', 'db', 'collection', 'operation']) {
+      it(`should respond with BSONRegExp class with option passed to ${passOptionTo}`, function(done) {
+        const client = this.configuration.newClient(passOptionTo === 'client' ? option : undefined);
+        let collection;
+
+        return client
```
Comment (dariakp) [inline test/unit/bson_regex.test.js:15]: since we are using done, this `return` should be removed, otherwise mocha gets confused

**Review by dariakp — CHANGES_REQUESTED** [review]

**Review by dariakp — APPROVED** [review]

**Review by emadum — APPROVED** [review]
LGTM 👍

**Pull request merged**